lib/compat/wordpress-6.8/ajax-actions.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Temporary compatibility shims for Core Ajax handlers.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Handles renewing the REST API nonce via AJAX.
+ *
+ * @since 5.3.0
+ * @since 6.8.0 Returns error when session token is missing.
+ */
+function gutenberg_ajax_rest_nonce() {
+	$token = wp_get_session_token();
+	if ( empty( $token ) ) {
+		wp_send_json_error( null, rest_authorization_required_code() );
```

Comment:
```suggestion
                $error = new WP_Error(
			'rest_unable_get_access_token',
			__( 'Sorry, unable to get session token.' ),
			array( 'status' => rest_authorization_required_code() ) 
		);
		wp_die(  $error );
```
I love to see a wp_die and wp_error combo here. wp_die is clever enough to use _ajax_wp_die_handler if it is admin ajax request.

Reply:
> I love to see a...

Any particular reason why this piece of code should be different/inconsistent with most other AJAX responses from admin-ajax? Don't think introducing inconsistencies is a good idea, sorry :) 

Also, as mentioned above I think this code may introduce edge cases as it is only checking the default cookies and not using the proper functionality?

Reply:
The `wp_send_json_error` uses `wp_die` internally.

See: https://developer.wordpress.org/reference/functions/wp_send_json/.

Reply:
Sure, let me reformulate the question: why not use wp_send_json_error() as most other WP code but use wp_die()? What advantages does that bring? Are the advantages worth it the change/inconsistency? (And other in the same line of thought) :)